code/game/objects/explosion_iterative.dm

```diff
@@ -0,0 +1,436 @@
+//an explosion reproducing a pretty "realist" impact with minimal load, very similar result to recursive, but without recursive function and also with only 1 pass on any tile (recursive would end up passing 2-3-4 times on most tiles covered by the explosion)
+/client/proc/kaboom()
+	var/power = input(src, "power?", "power?") as num
+	var/turf/T = get_turf(src.mob)
+	explosion_iter(T, power)
+
+/atom/var/explosion_resistance = 0.01 //everything soaks a tiny bit of damage at least
+
+
+
+
+proc/explosion_iter(turf/epicenter, devastation_range, heavy_impact_range, light_impact_range)
+	var/datum/line_explosion_controller/explosion = new()
+	explosion.startExplosion(epicenter, devastation_range, heavy_impact_range, light_impact_range)
+
+
+
+
+/obj/effect/debugging/marker/var/changedtimes = 0
+/obj/effect/debugging/marker/var/powerexplosion = 0
+/turf
+	var/explosion_ratio = 1.0 //this affect the explosion spread, but by a % instead of direct value (should make walls a bit more meaningfull vs stronger bombs
+	var/datum/side_explosion/side_explosion_data = 0
+
+/turf/space
+	explosion_resistance = 10
+
+/turf/simulated/floor
+	explosion_resistance = 1
+
+/turf/simulated/mineral
+	explosion_resistance = 2
+
+/turf/simulated/shuttle/floor
+	explosion_resistance = 1
+
+/turf/simulated/shuttle/floor4
+	explosion_resistance = 1
+
+/turf/simulated/shuttle/plating
+	explosion_resistance = 1
+
+/turf/simulated/shuttle/wall
+	explosion_resistance = 5
+	explosion_ratio = 0.80
+/turf/simulated/wall
+	explosion_resistance = 5
+	explosion_ratio = 0.80
+
+/turf/simulated/wall/r_wall
+	explosion_resistance = 15
+	explosion_ratio = 0.60
+
+//Code-wise, a safe value for power is something up to ~25 or ~30.. This does quite a bit of damage to the station.
+//direction is the direction that the spread took to come to this tile. So it is pointing in the main blast direction - meaning where this tile should spread most of it's force.
+
+/proc/dirtoangle(direction) //convert to angle starting from north
+{
```

Comment:
im not sure what your problem is with those variable definition, its definitly relative pathing since its relative to the type defined just above and has the indentation.